client.js

```diff
@@ -250,6 +266,74 @@ class Client {
     this.ws.send(buffer);
   }
 
+  /**
+   * Helper method to build a proper EventQuery message from a simple plain object.
+   *
+   *
+   *
+   *
+   * @param {Object} query - The simple plain object query.
+   * @returns {DBMessaging.Protobuf.EventQuery} - The built EventQuery message.
+   */
+  _buildEventQuery(query) {
+    const root = require('./generated/containerPb.js');
+    const { EventQuery } = root.DBMessaging.Protobuf;
+    const { MatchType } = EventQuery;
+
+    // Create base query with primitive fields
+    const baseQuery = {
+      timeRangeBegin: query.timeRangeBegin || 0,
+      timeRangeEnd: query.timeRangeEnd || Math.floor(Date.now() / 1000),
+      codeMask: query.codeMask !== undefined ? query.codeMask : 0xFFFFFFFF,
+      limit: query.limit || 50,
+      offset: query.offset || 0,
+      flags: query.flags || 0
+    };
+
+    // Build sender conditions if present
+    if (query.senders && query.senders.length > 0) {
+      baseQuery.senderConditions = {
+        conditions: query.senders.map(sender => ({
+          value: sender,
+          type: MatchType.Exact
```

Comment:
Hardcoding everything to MatchType.Exact seems like a step backwards. I think we should keep it configurable.